backend/src/schedule/schedule.service.ts

```diff
@@ -22,31 +23,49 @@ export class ScheduleService {
   }
 
   public async getScheduleById(id: string): Promise<ScheduleDTO> {
-    const schedule = await this.prismaService.schedule.findUniqueOrThrow({
-      where: { id },
-    });
+    try {
+      const schedule = await this.prismaService.schedule.findUniqueOrThrow({
+        where: { id },
+      });
 
-    return ScheduleService.convertToDTO(schedule);
+      return ScheduleService.convertToDTO(schedule);
+    } catch (e) {
+      if (e instanceof NotFoundError) {
+        throw new NotFoundException();
+      }
+    }
```

Comment:
the reason a `500` error was thrown before was because prisma does not have the same error handling as nest so we have to explicitly throw a `NotFoundException` (NestJS error) when prisma throws a `NotFoundError`